Tests/BluetoothTests/GATTCharacteristicTests.swift

```diff
@@ -607,52 +607,64 @@ final class GATTCharacteristicTests: XCTestCase {
         XCTAssertEqual(GATTCentralAddressResolution(data: Data([0x00])), false, "The value 0x00 should be interpreted as Not Supported")
     }
     
+
     func testBodyCompositionMeasurement() {
         
         let emptyUnit16 = UInt16(littleEndian: UInt16(bytes: (0x00, 0x00)))
         let data = Data([0x00, 0x00, 0x00, 0x00])
         
-        let SIMass = GATTBodyMass.init(rawValue: emptyUnit16, unit: MassUnit.init(rawValue: 0x2702)!)
-        let ImperialMass = GATTBodyMass.init(rawValue: emptyUnit16, unit: MassUnit.init(rawValue: 0x27B8)!)
+        typealias massUnit = GATTBodyCompositionMeasurement.MassUnit
+        typealias lengthUnit = GATTBodyCompositionMeasurement.LengthUnit
+        
+        let kilogram = massUnit(unit: .kilogram)
+        let pound = massUnit(unit: .pound)
+        let inch = lengthUnit(unit: .inch)
+        
+        
+        let SIMass = GATTBodyMass(rawValue: emptyUnit16, unit: massUnit(rawValue: 0x2702)!)
+        let ImperialMass = GATTBodyMass(rawValue: emptyUnit16, unit: massUnit(rawValue: 0x27B8)!)
         
-        let SILenght = GATTBodyLenght(rawValue: emptyUnit16, unit: LenghtUnit.init(rawValue: 0x2701)!)
-        let ImperialLenght = GATTBodyLenght(rawValue: emptyUnit16, unit: LenghtUnit.init(rawValue: 0x27A2)!)
+        let SILenght = GATTBodyLength(rawValue: emptyUnit16, unit: lengthUnit(rawValue: 0x2701)!)
+        let ImperialLenght = GATTBodyLength(rawValue: emptyUnit16, unit: lengthUnit(rawValue: 0x27A2)!)
         
         let Joule = GATTBodyEnergy(rawValue: emptyUnit16)
         let ohm = GATTBodyResistance(rawValue: emptyUnit16)
         
         let percentage = GATTBodyPercentage(rawValue: emptyUnit16)
         
+        let integerPercentage = GATTBodyPercentage(integerLiteral: 50)
+        
         let timeStamp = GATTDateTime(date: Date())
         let id = UInt8(bitPattern: 0x05)
         
-        let complexCharacteristic = GATTBodyCompositionMeasurement(bodyFatPercentage: percentage!, timeStamp: timeStamp, userIdentifier: id, basalMetabolism: Joule, musclePercentage: percentage!, muscleMass: ImperialMass, fatFreeMass: ImperialMass, softLeanMass: ImperialMass, bodyWaterMass: ImperialMass, impedance: ohm, weight: ImperialMass, height: ImperialLenght)
+        let complexCharacteristic = GATTBodyCompositionMeasurement(bodyFatPercentage: percentage, massUnit: pound!, lengthUnit: inch!, timeStamp: timeStamp, userIdentifier: id, basalMetabolism: Joule, musclePercentage: integerPercentage, muscleMass: ImperialMass, fatFreeMass: ImperialMass, softLeanMass: ImperialMass, bodyWaterMass: ImperialMass, impedance: ohm, weight: ImperialMass, height: ImperialLenght)
```

Comment:
@JurjDev avoid force wrapping variables by using guard and calling XCTAssertFailure if needed